### DISCUSSION TITLE
The cursor is indistinguishable between input mode and insert mode

### DISCUSSION THREAD

**Discussion by xlpxxx** [helix-editor/helix#10581] [General]
![1](https://github.com/helix-editor/helix/assets/46029824/133e45ed-35a5-4709-a17c-48df33f5baa2)
![2](https://github.com/helix-editor/helix/assets/46029824/91cde812-f4ee-4a47-8069-c464ba83d067)

Instead of having to check the status in the lower left corner every time, add an indicator at the cursor position.

**Comment by unwrinkled** [comment]
There is a configuration option where you can change the way the cursor looks depending on the mode. https://docs.helix-editor.com/configuration.html#editorcursor-shape-section

**Comment by pascalkuthe** [comment]
The cursor can also change color depending on the mode if the theme provides color for that.

**Comment by krish-r** [comment]
yeah, currently the following themes support mode-based cursor colours -

- ayu_evolve
- catppuccin_*
- cyan_light
- dracula
- onelight

(not an exhaustive list, there could be others themes)
